components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -37,11 +38,13 @@ public class LegacyJobOrchestrator : IJobOrchestrator, IPublisherConfigServices,
         /// <param name="agentConfigProvider">The provider that provides the agent configuration.</param>
         /// <param name="jobSerializer">The serializer to (de)serialize job information.</param>
         /// <param name="logger">Logger to write log messages.</param>
-        /// <param name="identity">Module's identity provider.</param>
-
+        /// <param name="publishedNodesProvider">Published nodes provider.</param>
+        /// <param name="jsonSerializer">Json serializer.</param>
         public LegacyJobOrchestrator(PublishedNodesJobConverter publishedNodesJobConverter,
             ILegacyCliModelProvider legacyCliModelProvider, IAgentConfigProvider agentConfigProvider,
-            IJobSerializer jobSerializer, ILogger logger, IIdentity identity) {
+            IJobSerializer jobSerializer, ILogger logger, PublishedNodesProvider publishedNodesProvider,
```

Comment:
>PublishedNodesProvider [](http://example.com/codeflow?start=58&length=22)

shouldn't we inject the interface here?

Reply:
Yes. I need to fix the interface, then I will change this to get it.

Reply:
Using interface.